session/manager_http.go

```diff
@@ -129,6 +131,7 @@ func (s *ManagerHTTP) IssueCookie(ctx context.Context, w http.ResponseWriter, r
 
 	cookie.Values["session_token"] = session.Token
 	cookie.Values["expires_at"] = session.ExpiresAt.UTC().Format(time.RFC3339Nano)
+	cookie.Values["nonce"] = randx.MustString(8, randx.Alpha) // Guarantee new kratos session identifier
```

Comment:
Can you add a test which executes this method with the same session and asserts that the cookie value is indeed different? Thanks!

Reply:
@aeneasr  I added a test to cover this actually. This does -

1. Create login flow
2. Submit Login
3. Validate the kratos session cookie exists 
4. Create force refresh login flow
5. Submit
6. Validate kratos session cookie exists
7. Validate both kratos session cookies are not the same value
8. Validate whoami session with both cookies sequentially

Does that work?

https://github.com/ory/kratos/blob/5be3aeb2eecc4bb2cb53863d57b3c9182478888f/selfservice/flow/login/handler_test.go#L239

Reply:
I saw that test - does it guarantee though that this is the case? What if for some reason a value in the cookie is different for this specific flow, and the underlying method is not doing what it should in the rest of cases? We need a dedicated test which checks for immutability here. The hook test is also fine, but it doesn't cover (for example) account recovery.

Reply:
Added one at `manager_http_test` - https://github.com/ory/kratos/pull/2761/commits/eea325558fb2a568a8e5442fb20eaa0f8aec70fa